pcli/src/command/stake.rs

```diff
@@ -242,17 +244,65 @@ impl StakeCmd {
-                        .unwrap_or(&BTreeMap::default())
-                        .values()
-                        .flat_map(|notes| notes.iter().map(|n| n.as_ref().amount()))
-                        .sum::<u64>(),
+                let mut unbonded = Value {
+                    amount: 0,
                     asset_id: *STAKING_TOKEN_ASSET_ID,
                 };
 
-                total += unbonded.amount;
+                let mut total_unbonding = Value {
+                    amount: 0,
+                    asset_id: *STAKING_TOKEN_ASSET_ID,
+                };
+                let mut unbonding_timeline = BTreeMap::new();
+
+                // Lifted out because borrow checker
+                let empty_notes = BTreeMap::default();
+
+                for note in notes
+                    .get(&*STAKING_TOKEN_DENOM)
+                    .unwrap_or(&empty_notes)
+                    .values()
+                    .flatten()
+                {
+                    // Add the note to the correct total, depending on whether it's quarantined currently
+                    if let UnspentNote::Quarantined {
+                        note,
+                        unbonding_height,
+                    } = note
+                    {
+                        total_unbonding.amount += note.amount();
+                        unbonding_timeline
+                            .entry(unbonding_height)
+                            .or_insert(Value {
+                                amount: 0,
+                                asset_id: *STAKING_TOKEN_ASSET_ID,
+                            })
+                            .amount += note.amount();
+                    } else {
+                        unbonded.amount += note.as_ref().amount();
+                    }
+                }
+
+                total += unbonded.amount + total_unbonding.amount;
+
+                // Separate out each chunk of unbonding stake into its own line item so we can
+                // display a separated ETA for each chunk of stake
+                let chain_height = state.last_block_height().unwrap_or_default();
+                for (unbonding_height, value) in unbonding_timeline {
+                    let remaining_blocks = unbonding_height - chain_height;
+                    const ESTIMATED_BLOCK_DURATION: Duration = Duration::from_secs(10);
+                    let eta = humantime::Duration::from(
+                        ESTIMATED_BLOCK_DURATION * remaining_blocks as u32,
+                    );
+                    let approx_eta = eta.to_string().split(' ').take(2).join(" ");
+
+                    table.add_row(vec![
+                        format!("Unbonding (ETA: {approx_eta})"),
+                        value.try_format(state.asset_cache()).unwrap(),
+                        format!("{:.4}", 1.0),
+                        value.try_format(state.asset_cache()).unwrap(),
+                    ]);
+                }
```

Comment:
Am I reading correctly that this code is only showing the estimated ETA (in units of time) and not the precise ETA (in blocks)?

I think it would be useful to display the exact unbonding time (in blocks), and then have the ETA in parentheses as an estimate afterwards.

That way, if the estimate bounces around, it's more clear that what's happening is that the *estimate* is moving, rather than the actual unbonding height.